src/conf_mode/interfaces-bonding.py

```diff
@@ -73,63 +73,83 @@ def get_config(config=None):
+    # later on for each member
     if 'member' in bond and 'interface' in bond['member']:
-        # convert list if member interfaces to a dictionary
-        bond['member']['interface'] = dict.fromkeys(
-            bond['member']['interface'], {})
+        # convert list of member interfaces to a dictionary
+        bond['member']['interface'] = {k: {} for k in bond['member']['interface']}
 
     if 'mode' in bond:
         bond['mode'] = get_bond_mode(bond['mode'])
 
     tmp = leaf_node_changed(conf, base + [ifname, 'mode'])
-    if tmp: bond.update({'shutdown_required': {}})
+    if tmp: bond['shutdown_required'] = {}
 
     tmp = leaf_node_changed(conf, base + [ifname, 'lacp-rate'])
-    if tmp: bond.update({'shutdown_required': {}})
+    if tmp: bond['shutdown_required'] = {}
 
     # determine which members have been removed
     interfaces_removed = leaf_node_changed(conf, base + [ifname, 'member', 'interface'])
+
+    # Reset config level to interfaces
+    old_level = conf.get_level()
+    conf.set_level(['interfaces'])
+
     if interfaces_removed:
-        bond.update({'shutdown_required': {}})
+        bond['shutdown_required'] = {}
         if 'member' not in bond:
-            bond.update({'member': {}})
+            bond['member'] = {}
 
         tmp = {}
         for interface in interfaces_removed:
             section = Section.section(interface) # this will be 'ethernet' for 'eth0'
-            if conf.exists(['insterfaces', section, interface, 'disable']):
-                tmp.update({interface : {'disable': ''}})
+            if conf.exists([section, interface, 'disable']):
+                tmp[interface] = {'disable': ''}
             else:
-                tmp.update({interface : {}})
+                tmp[interface] = {}
 
         # also present the interfaces to be removed from the bond as dictionary
-        bond['member'].update({'interface_remove': tmp})
+        bond['member']['interface_remove'] = tmp
+
+    # Restore existing config level
+    conf.set_level(old_level)
 
     if dict_search('member.interface', bond):
         for interface, interface_config in bond['member']['interface'].items():
+            # Check if member interface is a new member
+            if not conf.exists_effective(['member', 'interface', interface]):
+                bond['shutdown_required'] = {}
+
+            # Check if member interface is disabled
+            conf.set_level(['interfaces'])
+
+            section = Section.section(interface) # this will be 'ethernet' for 'eth0'
+            if conf.exists([section, interface, 'disable']):
+                interface_config['disable'] = ''
+
+            conf.set_level(old_level)
+
             # Check if member interface is already member of another bridge
             tmp = is_member(conf, interface, 'bridge')
-            if tmp: bond['member']['interface'][interface].update({'is_bridge_member' : tmp})
+            if tmp: interface_config['is_bridge_member'] = tmp
```

Comment:
That somehow never worked for me - operating on the reference. Are you sure it works now? That's why I stick with `bond['member']['interface'][interface]`

Reply:
As long as the reference itself is not replaced (i.e. `interface_config = ...`) this code should function the same (as they operate on the same reference). Just as a sanity check, I did print out `bond['member']['interface'][interface] is interface_config` just now in the for loop as a debugging measure and it returns `True`.

While I haven't checked individually if each key (e.g. 'is_bridge_member', 'is_source_address', etc) works, dumping the generated config during testing did show the config updating correctly.

Reply:
You can output the dict and see what happens by running „touch /tmp/vyos.ifconfig.debug“ and then commit

Reply:
I have already done something similar to that during testing (via insertion of print statements to file) but that's a good shortcut to know.